src/tests/fileIOSuite.c

```diff
@@ -0,0 +1,112 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#ifdef _WIN32
+#include <fcntl.h>
+#include <io.h>
+#endif
+
+#include <string.h>
+
+#include "CuTest.h"
+#include "fileIO.h"
+
+#define TEST_STRING "Lorem ipsum dolor sit amet, consectetur adipiscing elit.\nMorbi dignissim nunc vel lectus egestas, nec rutrum nibh gravida.\n\t(){}[];:<>\\\"'"
+
+/* When writing to a file and reading the file to a string, it should be the same */
+/* When writing a file, a file should be at the specified path */
+/* The file size should equate to the size of the string */
+/* When reading a file, the string size should equate to the file size */
+
+void test_read_write_roundtrip(CuTest *tc) {
+    const char *string = TEST_STRING;
+    char *read_string;
+
+    write_file("src/tests/testString.tmp", string);
+    read_string = read_file("src/tests/testString.tmp");
+
+    CuAssertStrEquals(tc, string, read_string);
+
+    free(read_string);
+}
+
+void test_write_file_exists(CuTest *tc) {
+    const char *string = TEST_STRING;
+    FILE *filePtr;
+
+    write_file("src/tests/testString2.tmp", string);
+    CuAssertTrue(tc, (filePtr = fopen("src/tests/testString2.tmp", "r")) != NULL);
+
+    fclose(filePtr);
+}
+
+void test_write_file_size(CuTest *tc) {
+    FILE *filePtr;
+    const char string[] = TEST_STRING;
+
+    write_file("src/tests/testString3.tmp", string);
+
+#ifdef _WIN32
+    _setmode(_fileno(filePtr), _O_BINARY);
+#endif
+    if ((filePtr = fopen("src/tests/testString3.tmp", "r")) == NULL) {
+        fprintf(stderr, "Error : could not open file");
+        exit(EXIT_FAILURE);
+    }
+
+    fseek(filePtr, 0, SEEK_END);
+    const size_t fileSize = ftell(filePtr);
+    fseek(filePtr, 0, SEEK_SET);
+
+    CuAssertTrue(tc, fileSize == strlen(string));
+
+#ifdef _WIN32
+    _setmode(_fileno(filePtr), _O_BINARY);
+#endif
+    fclose(filePtr);
+}
+
+void test_read_string_size(CuTest *tc) {

```

Comment:
For this method i would write the file as the first thing OR use the same file for all your methods.
It is unclear to me why this is the only test that doesn't write its own file. However you could totally just write one file and test that file throughout all the tests.

However; if you decide to only write one file make sure to write a comment in the get_file_io_suite function explaining that the order is important to ensure a file is written before it is read.

Reply:
made the function independent by writing a file in the start of the function which is then used to test.